test/xpath/test_base.rb

```diff
@@ -100,6 +100,12 @@ def test_root
       assert_equal "a", XPath::first( doc, "/" ).elements[1].name
     end
 
+    def test_nil_doc
+      assert_raise(NoMethodError) {
+        XPath.match(nil, '*')
+      }
+    end
```

Comment:
I don't think this test is needed.
I think it's fine if the wrong usage `XPath.match(anything_that_is_not_an_element, '*')` raises any kind of error or returns anything without error.

Reply:
Other part of this pull request looks good for to :+1:

Reply:
OK
I see.
Thansk!